src/App.js

```diff
@@ -1,18 +1,117 @@
 import React, { Component } from 'react';
 import styled from 'styled-components';
 import 'normalize.css';
+import SplitPane from 'react-split-pane';
+
+import Header from './components/Header';
+import Editor from './components/Editor';
+import UASTViewer from './components/UASTViewer';
+import languages from './languages';
 
 const Wrap = styled.div`
-  background-color: papayawhip;
-  color: palevioletred;
+  position: absolute;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
+  display: flex;
+  flex-direction: column;
+`
+
+const Content = styled.div`
+  display: flex;
+  height: 100%;
+  flex-direction: row;
+  position: relative;
 `
 
-class App extends Component {
+const initialState = {
+  languages: Object.assign({
+    auto: { name: '(auto)' },
+  }, languages),
+  actualLanguage: 'java',
```

Comment:
I can select `auto`, but bblfsh can tell it's `x` language.  It's basically for the auto. `selectedLanguage` overrides `actualLanguage` in all other cases.